kernel/src/platform/mod.rs

```diff
@@ -60,6 +60,43 @@ pub trait Platform {
     ) -> Result<(), returncode::ReturnCode> {
         Ok(())
     }
+
+    /// This function is called when an app faults
+    ///
+    /// This is an optional function that can be implemented by `Platform`s that
+    /// allows the chip to handle the app fault and not terminate or restart
+    /// the app.
+    ///
+    /// If `Ok(())` is returned by this function then the kernel will not
+    /// terminate or restart the app, but instead allow it to continue
+    /// running. NOTE in this case the chip must have fixed the underlying
+    /// reason for fault otherwise it will re-occur.
+    ///
+    /// This can not be used to get around Tock's protections. If for example
+    /// this function just ignored the error and allowed the app to continue
+    /// the fault would continue to occur,
```

Comment:
This is maybe too strong of a claim. A UKB implementation could somewhat easily end up running the next instruction after the one that caused the fault if they are not careful. Do we need to specify this?

Reply:
I have clarified this so that apps can't get around the protections. The `Platform` can avoid the protections